common/djangoapps/entitlements/api/v1/views.py

```diff
@@ -89,6 +89,27 @@ def _process_revoke_and_unenroll_entitlement(course_entitlement, is_refund=False
             raise IntegrityError
 
 
+def set_entitlement_policy(entitlement, site):
+    """
+    Assign the appropriate CourseEntitlementPolicy to the given CourseEntitlement based on its mode and site.
+
+    Arguments:
+        entitlement: Course Entitlement object
+        site: string representation of a Site object
+
+    Notes:
+        Site-specific, mode-agnostic policies take precedence over mode-specific, site-agnostic policies.
+        If no appropriate CourseEntitlementPolicy is found, the default CourseEntitlementPolicy is assigned.
+    """
+    policy_mode = entitlement.mode
+    if policy_mode == CourseMode.NO_ID_PROFESSIONAL_MODE:
+        policy_mode = CourseMode.PROFESSIONAL
+    filter_query = (Q(site=site) | Q(site__isnull=True)) & (Q(mode=policy_mode) | Q(mode__isnull=True))
+    policy = CourseEntitlementPolicy.objects.filter(filter_query).order_by('-site', '-mode').first()
```

Comment:
I believe order_by places null values at the beginning of the order. Would this mean that a null-site null-mode policy would be chosen over a more specific non-null mode/site policy?

Reply:
The minus sign in front of each of the attributes reverse orders them (so null values are at the end). That desired behavior of selecting the most specific available match over other applicable matches is verified in [this test](https://github.com/edx/edx-platform/pull/17758/files#diff-709d812e679055fa98e11ff10ec7f5d8R256).